robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java

```diff
@@ -1037,4 +1059,44 @@ public void onActivityDestroyed(Activity activity) {
       transcript.add("onActivityDestroyed");
     }
   }
+  
+  private static class ConfigChangingActivity extends TestActivity {
+    public int changingConfigurations;
+  
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+      assertTrue(isChangingConfigurations());
```

Comment:
Avoid putting assertions in test infrastructure. This would be better written if you step through the states in the test and then make the assertions in the test.